### PR TITLE
Initial commit of working version

### PR DESCRIPTION
**Pull request by jerryneedell**
This is my initial commit of a working version of the TMP007 driver.
I have tested it on a feather_m4_express

Comments welcome

### CODE REVIEW

File: adafruit_tmp007.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+        control &= ~(_TMP007_CFG_MODEON)
+        self._write_u16(_TMP007_CONFIG, control)
+
+    def wake(self):
+        """Wake up TMP007 from low power sleep mode."""
+        control = self._read_u16(_TMP007_CONFIG)
+        control |= _TMP007_CFG_MODEON
+        self._write_u16(_TMP007_CONFIG, control)
+
+    def read_raw_voltage(self):
```
Comment (tannewt) [inline adafruit_tmp007.py]: Please use properties for sensor values. I'd expect `temperature` for the object and `sensor_temperature` for the die temp. voltage can be private only.

File: adafruit_tmp007.py [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+
+
+    def __init__(self, i2c, address=_TMP007_I2CADDR):
+        """Initialize TMP007 device on the specified I2C address and bus number.
+        Address defaults to 0x40 and bus number defaults to the appropriate bus
+        for the hardware.
+        """
+        self._device = I2CDevice(i2c, address)
+
+    def begin(self, samplerate=CFG_16SAMPLE):
```
Comment (tannewt) [inline adafruit_tmp007.py]: Fold this into `__init__` so its going immediately. It can always be slept after.

**Review by tannewt — COMMENTED** [review]
Thanks! Just two small things.

File: adafruit_tmp007.py (hunk 1 above)
Comment (jerryneedell) [reply adafruit_tmp007.py]: made properties -- the voltage is not actually used internally. I think it is to be available for detailed sensor calibration by the user.

File: adafruit_tmp007.py (hunk 2 above)
Comment (jerryneedell) [reply adafruit_tmp007.py]: I  merged begin int init and just throw an error if the ID is not valid. Is this what you wanted?

File: adafruit_tmp007.py [hunk 3]
```diff
@@ -109,10 +106,11 @@ def begin(self, samplerate=CFG_16SAMPLE):
         config = _TMP007_CFG_MODEON | _TMP007_CFG_DRDYEN | samplerate
         self._write_u16(_TMP007_CONFIG, config)
         # Check device ID match expected value.
-        return  self.read_register(_TMP007_DEVID) == 0x0078
-
-
+        dev_id = self.read_register(_TMP007_DEVID)
+        if dev_id != 0x78:
+            raise ValueError('Init failed - Did not find TMP007')
```
Comment (tannewt) [inline adafruit_tmp007.py]: RuntimeError please. The argument could be valid but the circuit setup wrong.

File: adafruit_tmp007.py [hunk 4]
```diff
@@ -121,13 +119,15 @@ def sleep(self):
         control &= ~(_TMP007_CFG_MODEON)
         self._write_u16(_TMP007_CONFIG, control)
 
+    @property
```
Comment (tannewt) [inline adafruit_tmp007.py]: Sleep and wake should be functions because they are behavior rather than state.

**Review by tannewt — CHANGES_REQUESTED** [review]
Super close. Thanks Jerry!

File: adafruit_tmp007.py (hunk 3 above)
Comment (jerryneedell) [reply adafruit_tmp007.py]: Done

File: adafruit_tmp007.py (hunk 4 above)
Comment (jerryneedell) [reply adafruit_tmp007.py]: Done

**Review by tannewt — APPROVED** [review]
Great! Thank you!

**Pull request merged**